server/src/main/java/au/org/aodn/ogcapi/server/common/RestExtService.java

```diff
@@ -0,0 +1,195 @@
+                                    if (b.has("prefLabel") && b.has("_about")) {
+                                        CategoryVocabModel c = CategoryVocabModel
+                                                .builder()
+                                                .about(about.apply(b))
+                                                .label(label.apply(b))
+                                                .build();
+                                        broader.add(c);
+                                    }
+                                }
+                            }
+
+                            if (j.has("narrower")) {
+                                for (JsonNode b : j.get("narrower")) {
+                                    if (b.has("prefLabel") && b.has("_about")) {
+                                        CategoryVocabModel c = CategoryVocabModel
+                                                .builder()
+                                                .about(about.apply(b))
+                                                .label(label.apply(b))
+                                                .build();
+
+                                        narrower.add(c);
+
+                                        // The record comes from ardc have two levels only, so the second level for sure
+                                        // is empty, but the third level info comes form another link (aka the leaves)
+                                        // and therefore we can attach it to the second level to for the third.
+                                        if(leaves.containsKey(about.apply(b))) {
+                                            c.setNarrower(leaves.get(about.apply(b)));
+                                        }
+                                    }
+                                }
+                            }
+
+                            CategoryVocabModel model = CategoryVocabModel
+                                    .builder()
+                                    .label(label.apply(j))
+                                    .definition(definition.apply(j))
+                                    .about(about.apply(j))
+                                    .broader(broader)
+                                    .narrower(narrower)
+                                    .build();
+
+                            result.add(model);
+                        }
+                    }
+
+                    if (!node.isEmpty() && node.has("next")) {
+                        url = node.get("next").asText();
+                    }
+                    else {
+                        url = null;
+                    }
+                }
+                else {
+                    url = null;
+                }
+            } catch (RestClientException e) {
+                log.error("Fail connect {}, category return likely outdated", url);
+                url = null;
+            }
+        }
+
+        return result;
```

Comment:
the API seems to return excessive answers

as in current portal, every bottom-level vocab is under a second-level vocab; and every second-level vocab is under the top-level vocab;

this API currently returning a mixed levels for example:

```json
    {
        "label": "Air pressure",
        "definition": "This category contains vocabulary terms describing air pressure parameters",
        "about": "http://vocab.aodn.org.au/def/parameter_classes/category/10",
        "broader": [
            {
                "label": "Physical-Atmosphere",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/53"
            }
        ],
        "narrower": []
    },
```
understanding that above answer coming from: https://vocabs.ardc.edu.au/viewById/24 which doesn't have `narrower` for second-level vocab; the list of `narrower` assign to second-level vocab (in above example: `Air pressure`) through another mapping process, which will become this (which also returned in the same response with the above snippet):

```json
    {
        "label": "Physical-Atmosphere",
        "definition": "This category contains vocabulary terms describing physical atmosphere parameters",
        "about": "http://vocab.aodn.org.au/def/parameter_classes/category/53",
        "broader": [],
        "narrower": [
            {
                "label": "Air pressure",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/10",
                "narrower": [
                    {
                        "label": "Pressure (measured variable) exerted by the atmosphere and correction to sea level",
                        "definition": "Measurement as a phenomenon (as opposed to a co-ordinate) of the force per unit area exerted by the atmosphere determined in-situ at a known altitude and converted to the value at sea level assuming an isothermal layer.",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/77"
                    },
                    {
                        "label": "Pressure (measured variable) exerted by the atmosphere",
                        "definition": "Measurement as a phenomenon (as opposed to a co-ordinate) of the force per unit area exerted by the atmosphere determined in-situ at a known altitude.",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/CAPHZZ01"
                    }
                ]
            },
            {
                "label": "Visibility",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/22",
                "narrower": [
                    {
                        "label": "Horizontal visibility in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/390"
                    },
                    {
                        "label": "Present weather in the atmosphere",
                        "definition": "Visual observation using WMO code table 4677 or 4501",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/397"
                    }
                ]
            },
            {
                "label": "Air-Sea Fluxes",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/30",
                "narrower": [
                    {
                        "label": "Net shortwave heat flux",
                        "definition": "The net shortwave heat flux is the difference between the downwelling radiation from the sun as modified by the gases and clouds of the atmosphere and the radiation returned from the sea surface, which is either reflected by the surface or backscattered from within the water column.",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/10"
                    },
                    {
                        "label": "Net heat flux",
                        "definition": "Net surface heat flux across the air-sea interface.",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/11"
                    },
                    {
                        "label": "Net mass flux",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/12"
                    },
                    {
                        "label": "Downwelling vector irradiance as photons (PAR wavelengths) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/572"
                    },
                    {
                        "label": "Downwelling vector irradiance as energy (solar (300-3000nm) wavelengths) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/576"
                    },
                    {
                        "label": "Sensible heat flux",
                        "definition": "Heat exchange due to difference in temperature between air and water.",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/6"
                    },
                    {
                        "label": "Downwelling vector irradiance as energy in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/651"
                    },
                    {
                        "label": "Latent heat flux",
                        "definition": "Heat loss due to evaporation",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/7"
                    },
                    {
                        "label": "Sensible heat flux due to precipitation",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/8"
                    },
                    {
                        "label": "Net longwave heat flux",
                        "definition": "The net longwave heat flux is the net flux of the greybody emissions from the sea surface, cloud layers and the gases of the atmosphere.",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/9"
                    },
                    {
                        "label": "Downwelling vector irradiance as energy (longwave) in the atmosphere",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/LWRDZZ01"
                    }
                ]
            },
            {
                "label": "UV radiation",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/48",
                "narrower": [
                    {
                        "label": "Downwelling vector irradiance as energy (ultra-violet wavelengths) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/733"
                    }
                ]
            },
            {
                "label": "Humidity",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/8",
                "narrower": [
                    {
                        "label": "Dew point temperature of the atmosphere",
                        "definition": "The temperature to which air must cool to become saturated with water vapour",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/CDEWZZ01"
                    },
                    {
                        "label": "Specific humidity of the atmosphere",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/CHUMSS01"
                    },
                    {
                        "label": "Relative humidity of the atmosphere",
                        "definition": "The ratio of the amount of water vapour in the air compared to the maximum amount of water vapour that can theoretically be held at the air's temperature",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/CRELZZ01"
                    },
                    {
                        "label": "Wet bulb temperature of the atmosphere",
                        "about": "http://vocab.nerc.ac.uk/collection/P01/current/CWETZZ01"
                    }
                ]
            },
            {
                "label": "Precipitation and evaporation",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/9",
                "narrower": [
                    {
                        "label": "Thickness of precipitation amount (liquid water equivalent) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/386"
                    },
                    {
                        "label": "Precipitation duration (liquid water equivalent) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/405"
                    },
                    {
                        "label": "Hail duration in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/637"
                    },
                    {
                        "label": "Thickness of hail amount in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/638"
                    },
                    {
                        "label": "Hail intensity in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/639"
                    },
                    {
                        "label": "Precipitation rate (liquid water equivalent) in the atmosphere",
                        "about": "http://vocab.aodn.org.au/def/discovery_parameter/entity/94"
                    }
                ]
            }
        ]
    },
```
why would the consumer (or the portal) needs to know

```json
    {
        "label": "Air pressure",
        "definition": "This category contains vocabulary terms describing air pressure parameters",
        "about": "http://vocab.aodn.org.au/def/parameter_classes/category/10",
        "broader": [
            {
                "label": "Physical-Atmosphere",
                "about": "http://vocab.aodn.org.au/def/parameter_classes/category/53"
            }
        ],
        "narrower": []
    },
```

as a final result to the consumer; that is not a complete/accurate answer because `Air pressure` does have `narrower` not empty. I understand it was a correct result during processing (but only before adding related `narrower` and returning to the consumer);